packages/react-components/react-carousel-preview/library/src/components/CarouselCard/useCarouselCard.ts

```diff
@@ -14,52 +15,131 @@ import { useCarouselStore_unstable } from '../useCarouselStore';
-      let nextValue = currentIndex + 1 < snapshot.values.length ? snapshot.values[currentIndex + 1] : null;
-      let prevValue = currentIndex - 1 >= 0 ? snapshot.values[currentIndex - 1] : null;
+  const currentSelfIndex: number = useCarouselStore_unstable(snapshot => {
+    if (!snapshot.activeValue) {
+      return 0;
+    }
 
-      if (!nextValue && circular) {
-        nextValue = snapshot.values[0];
-      }
+    return snapshot.values.indexOf(value);
+  });
 
-      if (!prevValue && circular) {
-        prevValue = snapshot.values[snapshot.values.length - 1];
-      }
+  const totalCards: number = useCarouselStore_unstable(snapshot => snapshot.values.length);
 
-      if (nextValue === value || prevValue === value) {
-        return nextValue === value ? 'next' : 'prev';
-      }
+  const loopCount: number = useCarouselStore_unstable(snapshot => snapshot.loopCount);
+
+  // Track if we need to modify position due to circular loop
+  const cardDirection = currentActiveIndex < currentSelfIndex ? 'next' : 'prev';
+
+  // Nav Mod tracks cards that have switched their position due to circular
+  const navMod = navDirection === cardDirection ? 0.5 : -0.5;
+
+  // Direction mod tracks animation direction (shift 'forwards')
+  const directionMod = navDirection === 'next' ? 0.5 : -0.5;
+
+  let offsetIndex = circular ? loopCount * totalCards : 0;
+  if (circular && Math.abs(currentActiveIndex - (currentSelfIndex - directionMod)) + navMod >= totalCards / 2.0) {
+    offsetIndex = currentActiveIndex < currentSelfIndex ? offsetIndex - totalCards : offsetIndex + totalCards;
+  }
+
+  useIsomorphicLayoutEffect(() => {
+    // These cards will need to be positioned without animation
+    if (offsetIndex !== 0 && currentActiveIndex === 0) {
+      initialLoad.current = false;
     }
-  });
+  }, [currentActiveIndex, offsetIndex]);
+
+  if (currentActiveIndex !== 0) {
+    initialLoad.current = false;
+  }
+
+  const state: CarouselCardState = {
+    ...coreState,
+    offsetIndex,
+    initialLoad: initialLoad.current,
+  };
+
+  return state;
+};
+
+const useCarouselCardCore = (props: CarouselCardProps, ref: React.Ref<HTMLDivElement>): CarouselCardState => {
+  const { value } = props;
+  const { cardWidth } = useCarouselContext_unstable();
+  const [visible, setVisible] = React.useState(false);
+
+  const isActiveIndex = useCarouselStore_unstable(snapshot => snapshot.activeValue === value);
+
+  const offsetIndex = 0;
+
+  // Observe intersections of virtualized components
+  const { setObserverList } = useIntersectionObserver(
+    (entries: IntersectionObserverEntry[]) => {
+      // Grab latest entry that is intersecting
+      const latestEntry = entries[0];
+      if (!latestEntry) {
+        // If we don't find an intersecting area, ignore for now.
+        return;
+      }
+
+      // Set visible based on enter/exit ratio
+      setVisible(latestEntry.intersectionRatio >= 0.99);
+    },
+    {
+      threshold: 0.99,
```

Comment:
<img width="1496" alt="image" src="https://github.com/microsoft/fluentui/assets/14183168/f3f80702-0c7d-45b2-ac76-dd93ec43f584">

It's not a blocker, but it's probably something to consider as a bug...

Reply:
Good catch, we'll likely need to pass in the carousel container as the IO root so we only trigger on local visibility, we should be able to assign via context